### PR TITLE
feat(clock): Add relative monotonic clocks, platform detection

### LINKED ISSUE #62: Use monotonic relative time
Use relative time that starts at 0 instead of using system clock time.  Also, since the system clock isn't guaranteed to be monotonic, use performance.now or process.hrtime where available.

See: cujojs/most#452

### PR DESCRIPTION
**Pull request by briancavalier**
Close #62

Add relative monotonic clock implementations for performance.now and process.hrtime, and fallback clock for Date.now.  All clocks start at zero when created.

### Todo

- [x] Unit tests
- ~~Docs~~ will do in another PR

### CODE REVIEW

File: packages/scheduler/type-definitions/index.d.ts [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
 
 export function newScheduler (timer: Timer, timeline: Timeline): Scheduler;
 export function newScheduler (timer: Timer): (timeline: Timeline) => Scheduler;
 
 export function newDefaultScheduler (): Scheduler;
+
+export function newClockTimer (clock: Clock): Timer;
+export function newTimeline (): Timeline;
+
+declare function newPlatformClock (): Clock;
```
Comment (TylorS) [inline packages/scheduler/type-definitions/index.d.ts]: each `declare` should be changed to `export`

Comment (briancavalier) [inline packages/scheduler/type-definitions/index.d.ts]: Lol! I copied from flow and then proceeded to delete the wrong keyword from each line 🤣

Comment (briancavalier) [inline packages/scheduler/type-definitions/index.d.ts]: Fixed!

**Comment by TylorS** [comment]
I think this is good to go 👍

**Comment by briancavalier** [comment]
Cool.  Gonna write tests for the clock module and then I think this will be ready to merge.

**Comment by briancavalier** [comment]
At some point we need to tackle some minimal docs for `@most/scheduler`.  I'll try to do that in another PR.

File: packages/scheduler/src/clock.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    const hrt = this._hrtime(this.origin)
+    return (hrt[0] * 1e9 + hrt[1]) / 1e6
+  }
+}
+
+export const millisecondClockFromNow = now =>
+  new MillisecondClock(now, now())
+
+export const newPerformanceNowClock = () =>
+  millisecondClockFromNow(performance.now)
```
Comment (frangio) [inline packages/scheduler/src/clock.js]: This should be `performance.now.bind(performance)`. It's failing for me on Firefox.

**Pull request merged**